### PR TITLE
Add some type safety to the conversion pattern

### PR DESCRIPTION
**Pull request by lewing**
After finding another type mismatch in the existing providers I did this.

### CODE REVIEW

File: Agents/Xamarin.Interactive.Mac/UnifiedRepresentationProvider.cs [hunk 1]
```diff
@@ -109,49 +109,37 @@ public override bool ShouldReadMemberValue (IRepresentedMemberInfo representedMe
...
+                r => new CGRect (r.X, r.Y, r.Width, r.Height)))
                 return true;
-            }
 
-            if (TryFindMatchingRepresentation<CGPoint, Point> (
+            if (TryConvertRepresentation<CGPoint, Point> (
                 representedType,
                 representations,
-                out var point)) {
-                represented = new Point (
```
Comment (lewing) [inline Agents/Xamarin.Interactive.Mac/UnifiedRepresentationProvider.cs]: 🤷‍♂️

File: Agents/Xamarin.Interactive/Representations/RepresentationProvider.cs [hunk 2]
```diff
@@ -38,7 +39,25 @@ public virtual IEnumerable<object> ProvideRepresentations (object obj)
...
+                representations,
+                out var matchedRepresentation)) {
+                resolved = converter (matchedRepresentation);
+                return true;
+            }
+            resolved = default (TResolvedType);
+            return false;
+        }
+
+        bool TryFindMatchingRepresentation<TResolvedType, TRepresentationType> (
```
Comment (bojanrajkovic) [inline Agents/Xamarin.Interactive/Representations/RepresentationProvider.cs]: As discussed, this is public API, so leave it as `protected virtual` for now.

File: Agents/Xamarin.Interactive/Representations/RepresentationProvider.cs [hunk 3]
```diff
@@ -38,7 +39,25 @@ public virtual IEnumerable<object> ProvideRepresentations (object obj)
             return false;
         }
 
-        protected virtual bool TryFindMatchingRepresentation<TResolvedType, TRepresentationType> (
+        protected virtual bool TryConvertRepresentation<TResolvedType, TRepresentationType> (
```
Comment (bojanrajkovic) [inline Agents/Xamarin.Interactive/Representations/RepresentationProvider.cs]: I wonder if these generic parameters shouldn't be the other way around, like they are in the `converter` func.

**Comment by lewing** [comment]
I rebased this on top of #147 for... clarity?

**Comment by bojanrajkovic** [comment]
build

**Comment by abock** [comment]
Closing. We can dig this back up if/when we ever cycle back around.

**Pull request closed without merging**